Directory.Build.props

```diff
@@ -17,7 +17,6 @@
     <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)\cecil.snk</AssemblyOriginatorKeyFile>
     <DefineConstants>$(DefineConstants);NET_CORE</DefineConstants>
     <RootNamespace></RootNamespace>
-    <PackageId>$(MSBuildProjectName.Replace('Mono', 'Microsoft.DotNet'))</PackageId>
```

Comment:
@sbomer I know this was to be able to use Mono.Cecil.Overrides in dotnet/linker, with these changes this logic doesn't make much sense but at the same time, dotnet/linker is about to be retired. What do you think about this change?